File: src/org/kopi/vkopi/lib/ui/vaadin/addons/client/event/LoginWindowListener.java
/*
 * Copyright (c) 1990-2016 kopiRight Managed Solutions GmbH
 *
 * This library is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License version 2.1 as published by the Free Software Foundation.
 *
 * This library is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public
 * License along with this library; if not, write to the Free Software
 * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
 *
 * $Id$
 */

package org.kopi.vkopi.lib.ui.vaadin.addons.client.event;

import java.io.Serializable;

/**
 * The login window listener used to notify the application
 * that a user is going to be connected.
 */
public interface LoginWindowListener extends Serializable {

  /**
   * Fired on login action.
   * @param username The login username.
   * @param password The login password.
   * @param language The login language.
   */
  public void onLogin(String username, String password, String language);
}

File: src/org/kopi/vkopi/lib/ui/vaadinflow/event/LoginWindowListener.kt
/*
 * Copyright (c) 2013-2020 kopiLeft Services SARL, Tunis TN
 * Copyright (c) 1990-2020 kopiRight Managed Solutions GmbH, Wien AT
 *
 * This library is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License version 2.1 as published by the Free Software Foundation.
 *
 * This library is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public
 * License along with this library; if not, write to the Free Software
 * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
 */
package org.kopi.galite.ui.vaadin.event

import java.io.Serializable

/**
 * The login window listener used to notify the application
 * that a user is going to be connected.
 */
interface LoginWindowListener : Serializable {
  /**
   * Fired on login action.
   * @param username The login username.
   * @param password The login password.
   * @param language The login language.
   */
  fun onLogin(username: String, password: String, language: String)
}
